packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -1202,6 +1324,41 @@ export default function makeKernelKeeper(kernelStorage, kernelSlog) {
     return makeUpgradeID(nextID);
   }
 
+  function addTerminatedVat(vatID) {
+    if (!terminatedVats.includes(vatID)) {
+      terminatedVats.push(vatID);
+      kvStore.set(`vats.terminated`, JSON.stringify(terminatedVats));
+    }
+  }
+
+  function getFirstTerminatedVat() {
+    if (terminatedVats.length) {
+      return terminatedVats[0];
+    }
+    return undefined;
+  }
+
+  function deleteTerminatedVat(vatID) {
```

Comment:
Suggested clarifying rename:
```suggestion
  function forgetTerminatedVat(vatID) {
```